ValveResourceFormat/Resource/ResourceTypes/Mesh.cs

```diff
@@ -79,8 +86,72 @@ public static bool IsCompressedNormalTangent(IKeyValueCollection drawCall)
                 long flagsLong =>
                 // TODO: enum
                 (flagsLong & 2) == 2,
+                byte flagsByte =>
+                (flagsByte & 2) == 2,
                 _ => false
             };
         }
+
+        private static VBIB VBIBFromDATA(IKeyValueCollection data)
+        {
+            var VBIB = new VBIB();
+
+            var vertexBuffers = data.GetArray("m_vertexBuffers");
+            foreach (var vb in vertexBuffers)
+            {
+                var vertexBuffer = BufferDataFromDATA(vb);
+                
+                var decompressedSize = vertexBuffer.ElementCount * vertexBuffer.ElementSizeInBytes;
+                if (vertexBuffer.Data.Length != decompressedSize)
+                {
+                    vertexBuffer.Data = MeshOptimizerVertexDecoder.DecodeVertexBuffer((int)vertexBuffer.ElementCount, (int)vertexBuffer.ElementSizeInBytes, vertexBuffer.Data);
+                }
+                VBIB.VertexBuffers.Add(vertexBuffer);
+            }
+            var indexBuffers = data.GetArray("m_indexBuffers");
+            foreach (var ib in indexBuffers)
+            {
+                var indexBuffer = BufferDataFromDATA(ib);
+
+                var decompressedSize = indexBuffer.ElementCount * indexBuffer.ElementSizeInBytes;
+                if (indexBuffer.Data.Length != decompressedSize)
+                {
+                    indexBuffer.Data = MeshOptimizerIndexDecoder.DecodeIndexBuffer((int)indexBuffer.ElementCount, (int)indexBuffer.ElementSizeInBytes, indexBuffer.Data);
+                }
+
+                VBIB.IndexBuffers.Add(indexBuffer);
+            }
+
+            return VBIB;
+        }
+
+        private static VBIB.OnDiskBufferData BufferDataFromDATA(IKeyValueCollection data)
+        {
+            VBIB.OnDiskBufferData buffer = default(VBIB.OnDiskBufferData);
```

Comment:
Would prefer this to be `new VBIB()`, if VBIB.OnDiskBufferData is a class instead of struct this is null and throws a null reference exception next line.